src/plugins/discover/public/application/main/components/top_nav/discover_topnav.tsx

```diff
@@ -59,11 +56,11 @@ export const DiscoverTopNav = ({
   onFieldEdited,
   persistDataView,
   updateAdHocDataViewId,
-  adHocDataViewList,
-  savedDataViewList,
-  updateDataViewList,
 }: DiscoverTopNavProps) => {
   const history = useHistory();
+  const adHocDataViewList = useInternalStateSelector((state) => state.dataViewAdHocList);
+  const dataView = useInternalStateSelector((state) => state.dataView!);
+  const [refreshListId, setRefreshListId] = useState(0);
```

Comment:
Personally I prefer the `savedDataViews` prop since it was more clear why the data view picker was updating. Here's an idea I think could be helpful:
- Bring back the `savedDataViews` prop to unified search.
- Instead of maintaining a local `dataViewList` in `discover_main_route.tsx`, we could add `dataViewList` to the `InternalStateContainer` and update it in `loadDefaultOrCurrentDataView`.
- We can pass the `dataViewList` from the state container to unified search, then reload and update it in `updateDataViewList` when necessary.

I think this would be a clean approach since we'd be sharing the same `dataViewList` across Discover and wouldn't have to pass it around everywhere or worry about syncing with unified search. Plus it makes good use of the new state container system which I like. WDYT?